game/src/common/waypoints.rs

```diff
@@ -221,8 +218,42 @@ impl InputWaypoints {
 
     fn update_waypoints_drawable(&mut self, ctx: &mut EventCtx) {
         let mut batch = GeomBatch::new();
-        for waypt in &self.waypoints {
-            batch.append(waypt.geom.clone());
+        let total_waypoints = self.waypoints.len();
+        for (idx, waypt) in &mut self.waypoints.iter().enumerate() {
+            let geom = {
+                let wp = {
+                    match idx {
+                        0 => WaypointPosition::Start,
+                        idx if idx == total_waypoints - 1 => WaypointPosition::End,
+                        // technically this includes the case where idx >= total_waypoints which should hopefully never happen
+                        _ => WaypointPosition::Middle,
+                    }
+                };
+
+                let color = {
+                    match wp {
+                        WaypointPosition::Start => Color::GREEN,
+                        WaypointPosition::End => Color::RED,
+                        WaypointPosition::Middle => {
+                            [Color::BLUE, Color::ORANGE, Color::PURPLE][idx % 3]
+                        }
+                    }
+                };
+
+                let mut geom = GeomBatch::new();
+
+                geom.push(color, waypt.hitbox.clone());
+
+                let order = char::from_u32('A' as u32 + idx as u32).unwrap();
+                geom.append(
+                    Text::from(Line(format!("{}", order)).fg(Color::WHITE))
+                        .render(ctx)
+                        .centered_on(waypt.center),
+                );
+
+                geom
```

Comment:
Instead of this extra lexical scoping block `{}` thing, we can just do `batch.append(geom)` directly. Or even skip the intermediate `geom` and directly push onto the one final `batch`. I think that would be a little simpler to read through

Reply:
Updated.